File: LeetCode/majority_element.cpp
/*
Problem Link:
https://leetcode.com/problems/majority-element/description/
*/

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int majorityElement = nums[0];
        int majorityElementFreq = 1;
        int len = nums.size();
        for(int iter = 1;iter<len;iter++){
            if(nums[iter] != majorityElement){
                majorityElementFreq--;
                if(majorityElementFreq == 0){
                    majorityElement = nums[iter];
                    majorityElementFreq++;
                }
            }
            else{
                majorityElementFreq++;
            }
        }
        return majorityElement;
    }
};

File: LeetCode/majority_element.java
class Solution {
    public int majorityElement(int[] nums) {
        int element  = nums[0];
        int count = 1;
        for(int iter = 1;iter< nums.length; iter++){
            if(element != nums[iter]){
                count--;
                if(count == -1){
                    element = nums[iter];
                    count = 1;
                }
            }
            else
                count++;
        }
        return element;
    }
}